features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql

```diff
@@ -864,6 +864,7 @@ CREATE TABLE IDP_AUTHENTICATOR (
             NAME VARCHAR(255) NOT NULL,
             IS_ENABLED CHAR (1) DEFAULT '1',
             DISPLAY_NAME VARCHAR(255),
+            DEFINED_BY VARCHAR(255) NOT NULL,
```

Comment:
If we are defining this column as `NOT NULL`, have we planned how to handle the migration for the existing authenticators?

Reply:
The migration approach:
           Add the following line to the migration db schema to add a new column to the table and set SYSTEM as the column value for all the existing entries.

`ALTER TABLE IDP_AUTHENTICATOR ADD COLUMN DEFINED_BY VARCHAR(255) NOT NULL DEFAULT 'SYSTEM';`